_posts/2018-07-17-its-a-lot-more-snappy.md

```diff
@@ -45,7 +40,7 @@ async function transformPokemon(): Promise<void> {
 }
 ```
 
-Instead of storing the complete dataset in memory and using it in several different places. We pulled out only the data that was absolutely necessary for displaying in the UI. We also made sure that this array mapping operation only happened once every time the UPDATE_POKEMON action was dispatched to our redux store, instead of in a component render method. You'll notice, we also store a copy of this transformed array so that we don't have to retrieve it over the network the next time. Which means we can run the following when mounting our component.
+Instead of storing the complete dataset in memory and using it in several different places, we pulled out only the data that was absolutely necessary for displaying in the UI. We also made sure that this array mapping operation only happened once every time the UPDATE_POKEMON action was dispatched to our redux store, instead of in a component render method. You'll notice, we also store a copy of this transformed array so that we don't have to retrieve it over the network the next time. Which meant we could run the following when mounting our component.
```

Comment:
> Instead of storing the complete dataset in memory and using it in several different places, we pulled out only the data that was absolutely necessary for displaying in the UI.

`using it in several different places` - is this bit important?

Perhaps...
`Instead of storing the complete dataset in memory, we filtered it to only include the data necessary for the UI.`

Does that lose your message?

Reply:
`array mapping` - was this mapping or filtering; or both?  

Consider in relation with my suggestion above.  Consistent language is important to avoid confusing the reader.

Reply:
> We also made sure that this array mapping operation only happened once every time the UPDATE_POKEMON action was dispatched to our redux store, instead of in a component render method.

suggestion...
`We made sure the data mapping was performed once; when UPDATE_POKEMON action was dispatched to our redux store`.  

Would this lose valuable information?

Could add...
```Performing data manipulation during component a `render` ...<describe why this is bad >...```

Reply:
I would describe it as an array mapping. Filtering might imply that we were stripping elements from the array. That's not the case.